configs/projects/puppet-agent.rb

```diff
@@ -1,34 +1,63 @@
 project "puppet-agent" do |proj|
-  # Project level settings our components will care about
-  proj.setting(:install_root, "/opt/puppetlabs")
-  proj.setting(:prefix, File.join(proj.install_root, "puppet"))
+  platform = proj.get_platform
 
-  if platform.is_eos?
-    proj.setting(:sysconfdir, "/persist/sys/etc/puppetlabs")
-    proj.setting(:link_sysconfdir, "/etc/puppetlabs")
-  elsif platform.is_osx?
-    proj.setting(:sysconfdir, "/private/etc/puppetlabs")
+  # Project level settings our components will care about
+  # Windows has its own seperate layout
+  if platform.is_windows?
+    proj.setting(:install_root, "#{platform.drive_root}/ProgramFiles/PuppetLabs")
```

Comment:
The program files dirs are laid down during OS install so while it is _possible_ to make them not ~1 and ~2, it's such an extreme edge case.  Also it is possible to disable 8.3 name generation in NTFS but again, it's an edge case and _generally_ you wouldn't do it prior to the OS installation there fore the 8.3 names for prog file will still get generated.